### PR TITLE
Powershell updates

### PR DESCRIPTION
**Pull request by filizt**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs [hunk 1]
```diff
@@ -31,30 +31,31 @@ public class GetAzureRMBlueprint : BlueprintCmdletBase
...
+        public string ManagementGroupName { get; set; }
 
-        [Parameter(Mandatory = false, ParameterSetName = ParameterSetDefault)]
-        [Parameter(Mandatory = true, ParameterSetName = ParameterSetByVersion)]
-        [Parameter(Mandatory = true, ParameterSetName = ParameterSetLatestPublished)]
+        [Parameter(ParameterSetName = BlueprintDefinitionByVersion, Position = 2, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Blueprint definition version.")]
+        [Parameter(ParameterSetName = BlueprintDefinitionByName, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Blueprint definition name.")]
         [ValidateNotNullOrEmpty]
-        public string[] ManagementGroupName { get; set; }
+        public string BlueprintDefinitionName { get; set; }
```
Comment (haitch) [inline src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs]: I would keep use Name.

image you are typing a cmdlet, -Name is definitely better than -BlueprintDefinitionName.

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs [hunk 2]
```diff
@@ -31,30 +31,31 @@ public class GetAzureRMBlueprint : BlueprintCmdletBase
...
-        [Parameter(Mandatory = true, ParameterSetName = ParameterSetByVersion)]
+        [Parameter(ParameterSetName = BlueprintDefinitionByVersion, Position = 2, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Blueprint definition version.")]
         [ValidateNotNullOrEmpty]
-        public string BlueprintVersion { get; set; }
+        public string BlueprintDefinitionVersion { get; set; }
 
-        [Parameter(Mandatory = true, ParameterSetName = ParameterSetLatestPublished)]
-        public SwitchParameter LatestPublished { get; set; }
+        [Parameter(Mandatory = false, HelpMessage = "Blueprint definition's latest published version.")]
+        public SwitchParameter LatestPublishedVersion { get; set; }
```
[the remark this answers is not in the document]
Comment (filizt) [reply src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs]: Thanks!

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs [hunk 3]
```diff
@@ -27,55 +27,66 @@
...
             try
             {
                 switch (ParameterSetName)
                 {
-                    case ParameterSetDefault:
-                        HandleDefaultParameterSet();
+                    case ManagementGroupScope:
+                        if (string.IsNullOrEmpty(ManagementGroupName))
+                            foreach (var bpList in BlueprintClient.ListBlueprints(GetManagementGroupsForCurrentUser().ToList()))
+                                WriteObject(bpList);
```
Comment (haitch) [inline src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs]: I would be careful at here, what the expect out put and the actual output.

If I am the caller, I would expect
IEnumerable<Blueprint>
but the actual out put might be 
IEnumerable<IEnumerable<Blueprint>>

which might confusing to people.

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/RemoveAzureRMBlueprintAssignment.cs [hunk 4]
```diff
@@ -24,97 +24,45 @@
...
-        /// Perform the deletion of a blueprint assignment.
-        /// </summary>
-        /// <param name="subscription">The subscription ID for the assignment.</param>
-        /// <param name="name">The name of the assignment</param>
-        private void PerformDelete(string subscription, string name)
         {
             try
             {
-                if (ShouldProcess(subscription + "/" + name))
+                switch (ParameterSetName)
```
Comment (haitch) [inline src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/RemoveAzureRMBlueprintAssignment.cs]: InputObject seems got ignored here.

File: src/ResourceManager/Blueprint/Commands.Blueprint/Common/BlueprintClient.cs [hunk 5]
```diff
@@ -105,39 +108,49 @@ public async Task<PSPublishedBlueprint> GetLatestPublishedBlueprintAsync(string
...
+            }    
+        }
+
+        public IEnumerable<IEnumerable<PSBlueprint>> ListBlueprints(List<string> mgList)
+        {
+            foreach (var mgName in mgList)
+            {
+                var listResponse = blueprintManagementClient.Blueprints.List(mgName);
+
+                yield return listResponse.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName));
```
Comment (haitch) [inline src/ResourceManager/Blueprint/Commands.Blueprint/Common/BlueprintClient.cs]:
```suggestion
                foreach(var bp in listResponse.Select(bp => PSBlueprint.FromBlueprintModel(bp, mgName)))
                {
                    yield return bp;
                }
```

and then your function can return 
``` C#
public IEnumerable<PSBlueprint> ListBlueprints(List<string> mgList)
```

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/RemoveAzureRMBlueprintAssignment.cs (hunk 4 above)
Comment (filizt) [reply src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/RemoveAzureRMBlueprintAssignment.cs]: copy/paste error. Fixing

File: src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs [hunk 6]
```diff
@@ -27,48 +27,42 @@
 namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
 {
     [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintAssignment", SupportsShouldProcess = true)]
-    public class NewAzureRMBlueprintAssignment : BlueprintCmdletBase
+    public class NewAzureRmBlueprintAssignment : BlueprintCmdletBase
     {
+        #region Class Constants
+        // Parameter Set names
+        const string CreateUpdateBlueprintAssignment = "BlueprintAssignment";
```
[the remark this answers is not in the document]
Comment (filizt) [reply src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs]: We talked on this. If we want to support more parameter sets in the future, we can just add here with different set names.

**Pull request merged**